sunscreen_runtime/debugger-frontend/src/App.tsx

```diff
@@ -200,25 +276,69 @@ const App = () => {
           { source: 1, target: 2, directed: true, arrowhead: 'normal' }
         ]
       }
-      setGraph(lineNumber !== 1 ? graph : exGraph)
-    }, [setLine, setGraph]
+      setGraph(lineNumber !== 1 ? graph : dataToGraph(await fetch(`/sessions/${session}`).then(d => d.json())))
+    }, [setLine, setGraph, session]
   )
 
   const updateSelection = useCallback(
-    (selection, e) => {select(selection); console.log(selection.nodes?.values().next().value)}, [select]
+    async (selection, e) => {
+      select(selection); 
+      const node = selection.nodes?.values().next().value;
+      console.log(node)
+      if (node != null) {
+        if (session.split('_')[0] == "fhe") {
+          setInfo({
+            ...selection.nodes?.values().next().value, 
+            ...(await fetch(`sessions/${session}/${node.id}`).then(d => d.json())).Bfv,
+            stacktrace: filterStackTrace(await fetch(`sessions/${session}/stacktrace/${node.id}`).then(d => d.json()))
+          })
+        } else {
+          setInfo({
+            ...selection.nodes?.values().next().value, 
+            value: (await fetch(`sessions/${session}/${node.id}`).then(d => d.json())).Zkp,
+            stacktrace: filterStackTrace(await fetch(`sessions/${session}/stacktrace/${node.id}`).then(d => d.json()))
+          })
+        }
+        
+      } else {
+        setInfo({id: "no node selected"})
+      }
+    }, [select, session]
+  )
+
+  const updateSession = useCallback(
+    (event) => {
+      const newSession = event.target.value
+      
+      setSession(newSession)
+    }, [setSession]
   )
 
+  useEffect(() => {
+    const update = async () => {
+      const graph = await updateProblematicNodes(dataToGraph(await fetch(`/sessions/${session}`).then(d => d.json())))
```

Comment:
As discussed, this approach results in a 20s-30s slowdown for sudoku. You should call `setGraph(graph)` first with the data from the session fetch. _After that_, asynchronously, you should go through the nodes and update the problematic ones. That way you don't wait for all `N` requests to complete before showing the graph/code.